agent/util-scripts/pbench-tool-meister

```diff
@@ -67,6 +67,50 @@ tar_path = None
 # FIXME: The client response channel should be in a shared constants module.
 client_channel = "tool-meister-client"
 
+class ToolMetadata:
+    def __init__(self, redis_server = None, install_path = None):
+        if not redis_server and not install_path:
+            print("Defaulting to /opt/pbench-agent")
+            self.mode = "json"
+            self.json = Path("/opt/pbench-agent", "tool-scripts", "meta.json")
+        elif redis_server:
+            self.mode = "redis"
+            self.redis_server = redis_server
+        else:
+            self.mode = "json"
+            self.json = Path(install_path, "tool-scripts", "meta.json")
+        self.data = None
+
+    #TODO: ADD BETTER ERROR CHECKING TO THIS METHOD
+    def getFullData(self):
+        if self.data:
+            return self.data
+
+        if self.mode == "json":
+            with self.json.open("r") as json_file:
+                metadata = json.load(json_file)
+                self.data = metadata
+        elif self.mode == "redis":
+            meta_raw = self.redis_server.get("tool-metadata")
+            if meta_raw is None:
+                print('Metadata was never loaded.')
```

Comment:
Seems like this should be a log rather than just a print. Maybe even an exception, since we hopefully aren't going to try asking for redis data if it hasn't been loaded yet.